src/brod_group_subscriber_v2.erl

```diff
@@ -379,14 +378,29 @@ handle_info(_Info, State) ->
 %%--------------------------------------------------------------------
 -spec terminate(Reason :: normal | shutdown | {shutdown, term()} | term(),
                 State :: term()) -> any().
-terminate(_Reason, #state{workers = Workers}) ->
-  terminate_all_workers(Workers),
-  ok.
+terminate(_Reason, #state{workers = Workers,
+                          coordinator = Coordinator,
+                          group_id = GroupId
+                         }) ->
+  ok = terminate_all_workers(Workers),
+  ok = flush_offset_commits(GroupId, Coordinator).
 
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
 
+%% best-effort commits flush, this is a synced call,
+%% worst case scenario, it may timeout after 5 seconds.
+flush_offset_commits(GroupId, Coordinator) when is_pid(Coordinator) ->
+    case brod_group_coordinator:commit_offsets(Coordinator) of
+        ok -> ok;
+        {error, Reason} ->
+            ?BROD_LOG_ERROR("group_subscriber_v2 ~s failed to flush commits "
+                            "before termination", [GroupId, Reason])
+    end;
```

Comment:
Nit: looks like you have deeper indentation here than in the rest of the module.
LGTM otherwise.